avocado/plugins/xunit.py

```diff
@@ -36,7 +36,7 @@ class XUnitResult(Result):
 
     def _escape_attr(self, attrib):
         attrib = ''.join(_ if _ in self.PRINTABLE else "\\x%02x" % ord(_)
-                         for _ in str(attrib))
+                         for _ in astring.to_text(attrib))
```

Comment:
This probably introduces a bug.  I need to write a reproducer, but what tips me that this may be a bug is that the xunit XML file is supposed to be UTF-8:

https://github.com/avocado-framework/avocado/blob/master/avocado/plugins/xunit.py#L125

So, this may introduce characters encoded in a different default encoding.

Reply:
True